### ISSUE TITLE
BLE sensor state not updated

### ISSUE THREAD

**Issue by titiviking** [custom-components/ble_monitor#481]
Hi,

I have a Thermoplus sensors that is sending peroidic updates for temp & humidity.

Although the updates are received from the sensor (as seen in the logs:

`2021-08-22 10:33:48 DEBUG (MainThread) [custom_components.ble_monitor.device_tracker] 0 BLE ADV messages processed last 60 seconds for 0 device tracker device(s).
2021-08-22 10:34:15 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] Data binary sensor received: {'voltage': 2.822, 'temperature': 21.3125, 'humidity': 91.9375, 'battery': 82.2, 'rssi': -67, 'mac': '9A080000002C', 'type': 'Lanyard/mini hygrometer', 'packet': 'no packet id', 'firmware': 'Thermoplus', 'data': True}
2021-08-22 10:34:15 DEBUG (MainThread) [custom_components.ble_monitor.sensor] Data measuring sensor received: {'voltage': 2.822, 'temperature': 21.3125, 'humidity': 91.9375, 'battery': 82.2, 'rssi': -67, 'mac': '9A080000002C', 'type': 'Lanyard/mini hygrometer', 'packet': 'no packet id', 'firmware': 'Thermoplus', 'data': True}
2021-08-22 10:34:18 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] Data binary sensor received: {'voltage': 2.822, 'temperature': 21.3125, 'humidity': 91.9375, 'battery': 82.2, 'rssi': -71, 'mac': '9A080000002C', 'type': 'Lanyard/mini hygrometer', 'packet': 'no packet id', 'firmware': 'Thermoplus', 'data': True}
2021-08-22 10:34:18 DEBUG (MainThread) [custom_components.ble_monitor.sensor] Data measuring sensor received: {'voltage': 2.822, 'temperature': 21.3125, 'humidity': 91.9375, 'battery': 82.2, 'rssi': -71, 'mac': '9A080000002C', 'type': 'Lanyard/mini hygrometer', 'packet': 'no packet id', 'firmware': 'Thermoplus', 'data': True}
2021-08-22 10:34:39 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] 2 MiBeacon BLE ADV messages processed for 0 binary sensor device(s) total. Priority queue = 0
2021-08-22 10:34:39 DEBUG (MainThread) [custom_components.ble_monitor.sensor] 2 BLE ADV messages processed for 2 measuring device(s).
2021-08-22 10:34:39 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: main event_loop stopped, finishing
2021-08-22 10:34:39 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: Scanning will be restarted
2021-08-22 10:34:39 DEBUG (Thread-4) [custom_components.ble_monitor] 373 HCI events processed for previous period.
2021-08-22 10:34:39 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: Run
2021-08-22 10:34:39 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: connected to hci0
2021-08-22 10:34:39 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: start main event_loop
2021-08-22 10:34:48 DEBUG (MainThread) [custom_components.ble_monitor.device_tracker] 0 BLE ADV messages processed last 60 seconds for 0 device tracker device(s).
2021-08-22 10:35:39 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] 0 MiBeacon BLE ADV messages processed for 0 binary sensor device(s) total. Priority queue = 0
2021-08-22 10:35:39 DEBUG (MainThread) [custom_components.ble_monitor.sensor] 0 BLE ADV messages processed for 2 measuring device(s).
2021-08-22 10:35:39 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: main event_loop stopped, finishing
2021-08-22 10:35:39 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: Scanning will be restarted
2021-08-22 10:35:39 DEBUG (Thread-4) [custom_components.ble_monitor] 337 HCI events processed for previous period.
2021-08-22 10:35:39 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: Run
2021-08-22 10:35:39 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: connected to hci0
2021-08-22 10:35:39 DEBUG (Thread-4) [custom_components.ble_monitor] HCIdump thread: start main event_loop
2021-08-22 10:35:48 DEBUG (MainThread) [custom_components.ble_monitor.device_tracker] 0 BLE ADV messages processed last 60 seconds for 0 device tracker device(s).
2021-08-22 10:35:55 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] Data binary sensor received: {'voltage': 2.823, 'temperature': 21.3125, 'humidity': 91.875, 'battery': 82.3, 'rssi': -70, 'mac': '9A080000002C', 'type': 'Lanyard/mini hygrometer', 'packet': 'no packet id', 'firmware': 'Thermoplus', 'data': True}
2021-08-22 10:35:55 DEBUG (MainThread) [custom_components.ble_monitor.sensor] Data measuring sensor received: {'voltage': 2.823, 'temperature': 21.3125, 'humidity': 91.875, 'battery': 82.3, 'rssi': -70, 'mac': '9A080000002C', 'type': 'Lanyard/mini hygrometer', 'packet': 'no packet id', 'firmware': 'Thermoplus', 'data': True}
2021-08-22 10:35:58 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] Data binary sensor received: {'voltage': 2.823, 'temperature': 21.3125, 'humidity': 91.875, 'battery': 82.3, 'rssi': -71, 'mac': '9A080000002C', 'type': 'Lanyard/mini hygrometer', 'packet': 'no packet id', 'firmware': 'Thermoplus', 'data': True}
2021-08-22 10:35:58 DEBUG (MainThread) [custom_components.ble_monitor.sensor] Data measuring sensor received: {'voltage': 2.823, 'temperature': 21.3125, 'humidity': 91.875, 'battery': 82.3, 'rssi': -71, 'mac': '9A080000002C', 'type': 'Lanyard/mini hygrometer', 'packet': 'no packet id', 'firmware': 'Thermoplus', 'data': True}`

The state of the sensor in HA is not being updated.
![image](https://user-images.githubusercontent.com/73351833/130349486-73030d17-c6b9-4abb-b1bb-62e64501b1c9.png)

How can I force an update of the state on every packet received by the sensor (even if values are the same, I'd like it to be updated)

**Comment by Ernst79** [comment]
It should update every `period` (default 60 seconds). Are you sure this is the correct sensor. It seems to work fine in the logs.

**Comment by Ernst79** [comment]
Are you still facing this issue?

**Comment by titiviking** [comment]
Yes I am... The sensor from the integration is not being updated.

**Comment by Ernst79** [comment]
I need some more info, but I don't have a clue yet. Some questions:

- is it only the humidity sensor, or also other sensors (e.g. temperature)
- Do you have other devices, is it only one device that is giving issues or all? 
- Can you show the entities of this device (via the BLE monitor tile in the integration menu). sometimes a `_2` is added at the sensor names.

- You could try the following
- turn off discovery in the options, submit
- Remove the device via the options (type `-` at the MAC address of the device)
- Restart HA
- Delete the sensor entities one by one
- Turn on discovery

**Comment by titiviking** [comment]
I've attempted your process... I wasn't able to delete the device... My sensor remained in the list of devices of the integration, even after multiple reboots...


I've re-added the entities again... but issue seem to be the same again... last update was now again 10min ago... hence it does not align to the periodic 60s update it should have... The sensor is brand new so it sould work...

**Comment by Ernst79** [comment]
Can you make a printscreen of your settings and of the device screen in HA?

**Comment by titiviking** [comment]
![Screenshot_20210906-193627](https://user-images.githubusercontent.com/73351833/132249546-3e5cf504-64be-43f4-8b70-86c19e9125b1.jpg)
![Screenshot_20210906-193548](https://user-images.githubusercontent.com/73351833/132249550-c57d0329-59e3-4c06-a76b-64bf4491ec54.jpg)

**Comment by Ernst79** [comment]
Is the device called sleepy working ok? Could you try replacing the battery?

**Comment by titiviking** [comment]
The "sleepy" is the problematic one... It used to be called "bedroom" but wanted to try a different name for the purpose of troubleshooting. Device is brand new since few weeks with a new battery... From time to time I get new values in HA so device seems to broadcast correctly

**Comment by titiviking** [comment]
I can confirm the device is working properly indeed. I was able to retrieve all historical data through the "SensorBlue" app... Also I get proper live values in that app, indicating the sensor is working as expected

**Comment by Ernst79** [comment]
Does it improve if you move it closer to your HA machine?

**Comment by titiviking** [comment]
Yes I use an ssd attached to my raspberry... Otherwise I/O are too slow for the HA database

**Comment by Ernst79** [comment]
Ah, an SSD often interferes with the Bluetooth signal, causing serious reception issues. Try to place the SSD as far away from your raspberry with an extension cable or use an USB Bluetooth dongle on an usb extension cable, again placed as far away from the SSD as possible.

**Comment by Ernst79** [comment]
Did you try moving the SSD away? Any improvements?

**Comment by titiviking** [comment]
I did migrate my ssd to a new one that is further away, better quality and within a better enclosure.
Migration was done yesterday, for now everything seems to work fine and updates seem to come in regularly.

So looks like the problem is fixed (at least for now).
I also did upgrade of raspberry, which seem to resolve a few bugs in tte Bluetooth libraries (lost the Mac address of my BT adapter since previuos update).

Thanks for all the help.
